contentctl/objects/baseline_tags.py

```diff
@@ -17,6 +17,7 @@
 class BaselineTags(BaseModel):
     analytic_story: list[Story] = Field(...)
     #deployment: Deployment = Field('SET_IN_GET_DEPLOYMENT_FUNCTION')
+    # TODO (cmcginley): can we remove str from the possible types here?
     detections: List[Union[Detection,str]] = Field(...)
```

Comment:
This typing is weird, but intentional.  In the baseline files, we define the related searches initially as a list of strings.  Then, we construct the Baseline objects.  However, at this time, the related Detection objects do not exist!

Then, later, we construct the detections and iterate over the Baselines and convert the list of detection name strings into a list of Detection OBJECTS. You can see the code for this here:
https://github.com/splunk/contentctl/blob/96b813fa220ffef6349487421687b99404f647a3/contentctl/objects/abstract_security_content_objects/detection_abstract.py#L401-L415

You will notice that because Detections are constructed after Baselines, Macros, Lookups, etc, they don't have this weird double-typing:  https://github.com/splunk/contentctl/blob/96b813fa220ffef6349487421687b99404f647a3/contentctl/objects/abstract_security_content_objects/detection_abstract.py#L199

I would love to do this a different way, but can't think of it - I am very open to suggestions.

Reply:
Understood! Will investigate later as part of #223